src/controllers/popupEditorBackupController.js

```diff
@@ -1,10 +1,10 @@
-const { hasPermission } = require('../utilities/permissions');
+const helper = require('../utilities/permissions');
 
 const popupEditorBackupController = function (PopupEditorBackups) {
   const getAllPopupEditorBackups = function (req, res) {
     PopupEditorBackups.find()
-      .then(results => res.status(200).send(results))
-      .catch(error => res.status(404).send(error));
+      .then((results) => res.status(200).send(results))
+      .catch((error) => res.status(404).send(error));
   };
 
   const getPopupEditorBackupById = function (req, res) {
```

Comment:
The piece of code down here is incorrect:
```
try {
      PopupEditorBackups.find({ popupId: { $in: popupId } }, (error, popupBackup) => {
        res.status(200).send(popupBackup[0]);
      });
    } catch (error) {
      res.status(404).send(error);
    }
```
We pass a callback function as a second parameter, and while that is allowed, the error is not being accounted for. The exception you throw in your unit test actually is a programming error of doing: `null[0]`. This is wrong, and should be fixed. Replace with the following code instead:

```
try {
      const popupBackup = await PopupEditorBackups.findById(popupId);
      res.status(200).send(popupBackup);
    } catch (error) {
      res.status(404).send(error);
    }
```

make sure you update `const getPopupEditorBackupById = function (req, res) ` to   const `getPopupEditorBackupById = async function (req, res)` since we are using async await here.

The intention is to getPopupEditorBackup by id, so even using `find` is nonsensical. `FindById` is what you should use. More: https://mongoosejs.com/docs/api/model.html#Model.findById(). `find` actually returns an empty array if no documents match the query, and as a consequence `[][0]` returns undefined. The frontend should expect either a model or undefined.

Update test cases accordingly.